RIOT-OS/RIOT
# cpu/samd21: add support for SAMD20 & SAM D20 Xplained Pro board

### Contribution description

SAM D20 is a cut-down version of the SAM D21 MCU.
The reduced set of peripherals means

 - no DMA
 - no TCC timers
 - no fractional Baud for SERCOM
 - no I2C > 400 kHz
 - no USB

### Testing procedure

I verified that the configured peripherals are working.

I ran `./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . samd20-xpro --jobs=4 --with-test-only`:

```
Failures during test:
- [tests/driver_at86rf2xx_aes](tests/driver_at86rf2xx_aes/test.failed)
- [tests/emcute](tests/emcute/test.failed)
- [tests/gnrc_dhcpv6_client](tests/gnrc_dhcpv6_client/test.failed)
- [tests/gnrc_dhcpv6_client_6lbr](tests/gnrc_dhcpv6_client_6lbr/test.failed)
- [tests/gnrc_ipv6_ext](tests/gnrc_ipv6_ext/test.failed)
- [tests/gnrc_ipv6_ext_frag](tests/gnrc_ipv6_ext_frag/test.failed)
- [tests/gnrc_ipv6_ext_opt](tests/gnrc_ipv6_ext_opt/test.failed)
- [tests/gnrc_ipv6_nib_dns](tests/gnrc_ipv6_nib_dns/test.failed)
- [tests/gnrc_rpl_srh](tests/gnrc_rpl_srh/test.failed)
- [tests/gnrc_sock_dns](tests/gnrc_sock_dns/test.failed)
- [tests/gnrc_tcp](tests/gnrc_tcp/test.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.failed)
- [tests/pkg_libfixmath_unittests](tests/pkg_libfixmath_unittests/test.failed)
- [tests/shell](tests/shell/test.failed)
```

No surprises except for `tests/shell` which prints

```
socat - open:/dev/ttyACM0,b115200,echo=0,raw 

> bufsize

> bufsize
128
________________________________________________________________________________________________________________________________verylong
> ________________________________________________________________�	�.�)t��/�rCkk�
> ��+�zٗ�H�>�0a&7�'�9"_g�g,6�Ž�Timeout in expect script at "child.expect('shell: maximum line length exceeded')" (tests/shell/tests/01-run.py:163)

make: *** [/home/benpicco/dev/RIOT/tests/shell/../../Makefile.include:761: test] Error 1
``` 

But the shell is working correctly, so I think this is a fluke.

### Issues/PRs references

- PWM support requires #14478
